python/tutorials/06-fused-attention.py

```diff
@@ -497,38 +493,77 @@ def forward(ctx, q, k, v, causal, sm_scale):
 
     @staticmethod
     def backward(ctx, do):
+        # configuration is not supported
+        assert(not (ctx.split_kernel and not ctx.causal))
         if torch.version.hip is not None:
             BLOCK = 64
         else:
             BLOCK = 128
         q, k, v, o, L = ctx.saved_tensors
         do = do.contiguous()
-        dq = torch.zeros_like(q)
+        dq = torch.zeros_like(q, dtype=torch.float32)
```

Comment:
I'm not sure what the source of this was - I saw it in the original code as well - but I think q dtype can be fp16. Especially with block ptrs, this doesn't work and throws an error since the block is fp16.

Reply:
This came from current upstream: https://github.com/openai/triton/blob/main/python/tutorials/06-fused-attention.py#L279

> throws an error since the block is fp16
Yep, I've added separate initialization on line 538, but not sure if this is the best solution

Reply:
> I think it is the best solution for now

Well, we need to check I think.
Upstream optimizing for NV, it could work differently in our case.